packages/passport-interface/src/FeedAPI.ts

```diff
@@ -1,31 +1,22 @@
 import { PCDPackage } from "@pcd/pcd-types";
-import { FeedRequest, FeedResponse, ListFeedsResponse } from "./RequestTypes";
+import { ListFeedResult, requestListFeed } from "./api/requestListFeed";
+import { PollFeedResult, requestPoolFeed } from "./api/requestPoolFeed";
+import { FeedRequest } from "./RequestTypes";
 
 export interface IFeedApi {
-  pollFeed(providerUrl: string, request: FeedRequest): Promise<FeedResponse>;
-  listFeeds(providerUrl: string): Promise<ListFeedsResponse>;
+  pollFeed(providerUrl: string, request: FeedRequest): Promise<PollFeedResult>;
+  listFeeds(providerUrl: string): Promise<ListFeedResult>;
 }
 
 export class NetworkFeedApi implements IFeedApi {
   async pollFeed<T extends PCDPackage>(
     providerUrl: string,
     request: FeedRequest<T>
-  ): Promise<FeedResponse> {
-    const result = await fetch(providerUrl, {
-      method: "POST",
-      body: JSON.stringify(request),
-      headers: {
-        "Content-Type": "application/json",
-        Accept: "application/json"
-      }
-    });
-    const parsed = (await result.json()) as FeedResponse;
-    return parsed;
+  ): Promise<PollFeedResult> {
+    return requestPoolFeed(providerUrl, request);
   }
 
-  async listFeeds(providerUrl: string): Promise<ListFeedsResponse> {
-    const result = await fetch(providerUrl);
-    const parsed = (await result.json()) as ListFeedsResponse;
-    return parsed;
+  async listFeeds(providerUrl: string): Promise<ListFeedResult> {
```

Comment:
nit: why is there a naming change here between feed (singular) and feeds (plural)?

Reply:
still need to fix this one

Reply:
fixed